ethereum/contracts/ERC20App.sol

```diff
@@ -110,16 +113,30 @@ contract ERC20App is AccessControl {
         address _token,
         address _sender,
         bytes32 _recipient,
-        uint256 _amount
+        uint256 _amount,
+        uint32 _paraId
     ) private pure returns (bytes memory) {
-        return
-            abi.encodePacked(
+        if(_paraId == 0) {
```

Comment:
question - should we handle these 2 cases differently at the application level here, or rather just at the dispatch level when received by our parachain?

Reply:
i guess it's cheaper to handle it here?

Reply:
I am not sure if it would be cheaper. My guess is that handling this in rust/parachain would be cheaper since rust has more opportunity for optimization given its static nature (this would depend on benchmarks to calculate weight). I think using optional types on the rust end buys us cleaner API(and code) for the mint call on the parachain side.

Reply:
yeah, Its not really a matter of performance/cost, just about a cleaner API as Alistair says.